eclipse-ee4j/krazo#282: Added HttpCommunicationWrapper for Liberty JEE9

jesse-gallagher opened the pull request:
This change should fix the trouble observed in https://github.com/eclipse-ee4j/krazo/issues/282 , allowing Krazo to work again on Liberty JEE 9 betas.

The test suites for WildFly and GlassFish should still pass without issue, but there's one test failure in Liberty. I believe it's unrelated to this change, though, and I'll create another issue to identify the trouble there.

erdlet commented on resteasy/src/main/java/org/eclipse/krazo/resteasy/core/LibertyHttpCommunicationUnwrapper.java:
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019 Eclipse Krazo committers and contributors
```
Can you please adapt the date?

erdlet commented on resteasy/src/main/java/org/eclipse/krazo/resteasy/core/LibertyHttpCommunicationUnwrapper.java:
```diff
@@ -0,0 +1,73 @@
+      Object requestState = getInstance.invoke(null, false);
+      Method getCurrentThreadsIExtendedRequest = requestStateClass.getDeclaredMethod("getCurrentThreadsIExtendedRequest"); //$NON-NLS-1$
+      return (HttpServletRequest)getCurrentThreadsIExtendedRequest.invoke(requestState);
+    } catch (Throwable e) {
```
Do we really need to catch Throwable? It's known as an antipattern to me. Isn't catching `Exception` enough?

erdlet reviewed (approved):
Looks good to me. Just two little comments which aren't critical.

erdlet commented:
Thanks @jesse-gallagher for submitting this PR! :)

jesse-gallagher commented:
Done on both counts. I bumped the copyright year, changed the `catch` to only catch explicitly-thrown exceptions, and also added some `requireNonNull` lines on retrieved objects with explanatory messages.

chkal commented:
@jesse-gallagher I'm a bit late, but anyway: Thanks for contributing this patch! :beers:
